lib/puppet/file_system/windows.rb

```diff
@@ -106,6 +106,10 @@ def chmod(mode, path)
     Puppet::Util::Windows::Security.set_mode(mode, path.to_s)
   end
 
+  def read(path)
```

Comment:
So I ran this locally and it results in thousands of test failures. I believe the problem is that a lot of existing code falls through here - and previously went to the default implementation at:

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/file_system/file_impl.rb#L79-L81

```ruby
def read(path, opts = {})
  path.read(opts)
end
```

The `path` object in this case is a `Pathname` instance. It *might* work if you adjust it to not ignore the user-supplied `opts`, but I'd still be a bit leery of making such a change globally as so many different calls flow down to this point to read files:

```ruby
def read(path, opts = {})
  if opts == {}
    path.read(opts) 
  else
    read_preserve_line_endings(path)
  end
end
```

I'd also be careful of the performance implications of making this change globally. Is there a way that we could instead update the few callsites that need this new behavior?